src/components/panel/panel.js

```diff
@@ -832,6 +850,47 @@ MdPanelRef.prototype._addStyles = function() {
 };
 
 
+/**
+ * Add ARIA-specific attributes appropriate for panels.
+ */
+MdPanelRef.prototype._configureAria = function() {
+  this._panelEl.attr('tabIndex', '-1');
+
+  this._configureTrapFocus();
+};
+
+
+/**
+ * Setup the focus traps.
+ * @private
+ */
+MdPanelRef.prototype._configureTrapFocus = function() {
+  if (this._config.trapFocus) {
+    var element = this._panelEl;
+    // Set up elements before and after the panel to capture focus and
+    // redirect back into the panel.
+    this._topFocusTrap = document.createElement('div');
+    this._topFocusTrap.classList.add('_md-panel-focus-trap');
+    this._topFocusTrap.tabIndex = 0;
+
+    this._bottomFocusTrap = this._topFocusTrap.cloneNode(false);
+
+    // When focus is about to move out of the panel, we want to intercept it and redirect it
+    // back to the panel element.
+    var focusHandler = function () {
+      element.focus();
```

Comment:
Might be worth mentioning that this _wraps_ the bottom from the bottom, but simply "sticks" it when shift-tabbing, which isn't the _best_ approach.

(I recently had someone tell me that the only perfect way of focus trapping is to actually set the `tabIndex` on everything _outside_ the trapped area to `-1` and then set it back when you're done trapping).